back-end/objects/RImageFolder.py

```diff
@@ -250,16 +250,22 @@ def __init__(
     def add_records(self, records: List[Tuple[str, int]], correct: bool):
         buffer = self.buffer_correct if correct else self.buffer_incorrect
         next_record = self.next_correct if correct else self.next_incorrect
-
         paths = [to_unix(record[0]) for record in records]
 
         # 1. update db with the result
-        db_update_many_by_paths(self.db_conn, self.table_name, paths, ('classified',),
-                                [(correct,) for _ in paths])
+        if correct:
+            value_list = [(self.CLS_CORRECT,) for _ in paths]
+        else:
+            value_list = [(self.CLS_INCORRECT,) for _ in paths]
+        db_update_many_by_paths(self.db_conn, self.table_name, paths, ('classified',), value_list)
 
         # 2. update buffer
-        buffer.extend(records)
-
+        if buffer == []:
+            buffer.extend(records)
+        else:
+            dic = dict(buffer)
+            dic.update(dict(records))
+            buffer = list(dic.items())
```

Comment:
There seems to be an remaining issue:

Let's say at the beginning `self.buffer_incorrect` stores images A, B, C, and `self.buffer_correct` stores images D, E. Now we update the model (e.g., train the model for a few epochs) and test the model again, and find out it predicts correctly on images A, B, D, E, and incorrectly on C.

Now the code will call `add_records( [A, B, D, E], True )` and `add_records( [C], False)`, and we will end up having [A, B, C, D, E] in `self.buffer_correct` and [C, D, E] in `self.buffer_incorrect`.

One potential solution is to change the function signature to `add_records(self, correct_records, incorrect_records)`, and we only call this function once after testing. Inside the function, we can implement an O(n) algorithm to update the buffers and avoid duplication.